django/utils/text.py

```diff
@@ -64,6 +67,76 @@ def _generator():
     return "".join(_generator())
 
 
+class DjangoHTMLParser(HTMLParser):
+    class TruncationCompleted(Exception):
+        pass
+
+    void_elements = {  # https://html.spec.whatwg.org/#void-elements
+        "area",
+        "base",
+        "br",
+        "col",
+        "embed",
+        "hr",
+        "img",
+        "input",
+        "link",
+        "meta",
+        "source",
+        "track",
+        "wbr",
+    }
```

Comment:
We have a subclass of `HTMLParser` in `django.test`:

https://github.com/django/django/blob/32d4b61c313be5169137047e9fb3668da20a5d89/django/test/html.py#L202-L222

Can we reuse it? or create a class/mixin they can share?